### PR TITLE
Added theming support

### PR DESCRIPTION
**Pull request by javierdlg**
## Summary of the Pull Request
Adds a theming API endpoint for font colors, size, and style. Also allows setting the default background and foreground.

### CODE REVIEW

File: src/cascadia/PublicTerminalCore/HwndTerminal.hpp [hunk 1]
```diff
@@ -4,34 +4,43 @@
...
+    EmptyBox = 4,
+};
+
+typedef struct _TerminalTheme
+{
+    COLORREF DefaultBackground;
+    COLORREF DefaultForeground;
+    CursorStyle CursorStyle;
+    COLORREF ColorTable[16];
+} TerminalTheme, *LPTerminalTheme;
```
Comment (dgriffen) [inline src/cascadia/PublicTerminalCore/HwndTerminal.hpp]: Why are you defining a special pointer type alias?

File: src/cascadia/WpfTerminalControl/TerminalTheme.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Terminal.Wpf
+{
+    /// <summary>
+    /// Structure for color handling in the terminal.
+    /// </summary>
+    /// <remarks>Pack = 1 removes the padding added by some compilers/processors for optimization purposes.</remarks>
+    [StructLayout(LayoutKind.Sequential)]
+    public class TerminalTheme
```
Comment (dgriffen) [inline src/cascadia/WpfTerminalControl/TerminalTheme.cs]: Shouldn't this be a `struct` not a `class`?

File: src/cascadia/PublicTerminalCore/HwndTerminal.hpp (hunk 1 above)
Comment (javierdlg) [reply src/cascadia/PublicTerminalCore/HwndTerminal.hpp]: While talking to richard he suggested we have a pointer and a regular typedef for the theme struct. He argued that it's best to have and not use it :p

File: src/cascadia/WpfTerminalControl/TerminalTheme.cs (hunk 2 above)
Comment (javierdlg) [reply src/cascadia/WpfTerminalControl/TerminalTheme.cs]: switched it to a struct after talking offline

File: src/cascadia/PublicTerminalCore/HwndTerminal.hpp [hunk 3]
```diff
@@ -4,34 +4,43 @@
...
-    __declspec(dllexport) HRESULT _stdcall StartSelection(void* terminal, COORD cursorPosition, bool altPressed);
-    __declspec(dllexport) HRESULT _stdcall MoveSelection(void* terminal, COORD cursorPosition);
-    __declspec(dllexport) void _stdcall ClearSelection(void* terminal);
-    __declspec(dllexport) const wchar_t* _stdcall GetSelection(void* terminal);
-    __declspec(dllexport) bool _stdcall IsSelectionActive(void* terminal);
-    __declspec(dllexport) void _stdcall DestroyTerminal(void* terminal);
-    __declspec(dllexport) void _stdcall RegisterWriteCallback(void* terminal, void __stdcall callback(wchar_t*));
-    __declspec(dllexport) void _stdcall SendKeyEvent(void* terminal, WPARAM wParam);
-    __declspec(dllexport) void _stdcall SendCharEvent(void* terminal, char16_t ch);
+enum class CursorStyle : int32_t
```
[the remark this answers is not in the document]
Comment (javierdlg) [reply src/cascadia/PublicTerminalCore/HwndTerminal.hpp]: Added cursor support for theming

**Pull request merged**